artemis-junit/artemis-junit-commons/src/main/java/org/apache/activemq/artemis/junit/EmbeddedActiveMQDelegate.java

```diff
@@ -626,43 +625,51 @@ public void sendMessage(SimpleString address, ClientMessage message) {
-                                                           amqEx);
+         EmbeddedActiveMQResourceException failureCause = null;
+
+         try (ClientConsumer consumer = session.createConsumer(address, browseOnly)) {
+            ClientMessage message = null;
+            if (timeout > 0) {
+               try {
+                  message = consumer.receive(timeout);
+               } catch (ActiveMQException amqEx) {
+                  failureCause =  new EmbeddedActiveMQResourceException(String.format("ClientConsumer.receive( timeout = %d ) for %s failed",
+                                                                        timeout, address.toString()), amqEx);
+               }
+            } else if (timeout == 0) {
+               try {
+                  message = consumer.receiveImmediate();
+               } catch (ActiveMQException amqEx) {
+                  failureCause = new EmbeddedActiveMQResourceException(String.format("ClientConsumer.receiveImmediate() for %s failed",
+                                                                       address.toString()), amqEx);
+               }
+            } else {
+               try {
+                  message = consumer.receive();
+               } catch (ActiveMQException amqEx) {
+                  failureCause = new EmbeddedActiveMQResourceException(String.format("ClientConsumer.receive() for %s failed",
+                                                                       address.toString()), amqEx);
+               }
             }
-         } else if (timeout == 0) {
-            try {
-               message = consumer.receiveImmediate();
-            } catch (ActiveMQException amqEx) {
-               throw new EmbeddedActiveMQResourceException(String.format("ClientConsumer.receiveImmediate() for %s failed",
-                                                                         address.toString()),
-                                                           amqEx);
+
+            if (message != null) {
+               try {
+                  message.acknowledge();
+               } catch (ActiveMQException amqEx) {
+                  failureCause = new EmbeddedActiveMQResourceException(String.format("ClientMessage.acknowledge() for %s from %s failed",
+                                                                       message, address.toString()), amqEx);
+               }
             }
-         } else {
-            try {
-               message = consumer.receive();
-            } catch (ActiveMQException amqEx) {
-               throw new EmbeddedActiveMQResourceException(String.format("ClientConsumer.receive() for %s failed",
-                                                                         address.toString()),
-                                                           amqEx);
+
+            return message;
+         } catch (ActiveMQException amqEx) {
```

Comment:
It only seems to look at and throws 'failureCause' if it catches ActiveMQException in the outer catch wrapper. All the inner catch wrappers swallow ActiveMQException while setting failureCause, meaning that failureCause then wont be looked at since if they failed 'message' will just be null and the overall method will have successfully returned null just above this.